cmd/generate/config/rules/kubernetes.go

```diff
@@ -0,0 +1,62 @@
+package rules
+
+import (
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+// The kubernetes rules are split into two functions to make the complex proximity matching of the data-key and the kind-identifier more readable and testable
+
+// KubernetesSecretWithDataBefore validates if we detected a kubernetes secret which contains data, before the resource identifier!
+func KubernetesSecretWithDataBefore() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "kubernetes-secret-with-data-before",
+		Description: "Possible Kubernetes Secret detected, posing a risk of leaking credentials/tokens from your deployments",
+		// We try to match secrets by looking if we have the keyword
+		Regex: generateUniqueTokenRegex(`(?i)(?:\b(data)\b)\W+(?:\w+\W+){0,200}?\bkind:\s*Secret\b.*`, true),
+
+		Keywords: []string{
+			"Secret",
+		},
+		// Kubernetes secrets are always yaml files, we limit to common yaml-endings to make this rule more safe!
+		Path: regexp.MustCompile(`(?i)\.ya?ml$`),
+	}
+
+	// validate
+	tps := map[string]string{
+		// Sample Kubernetes Secret from https://kubernetes.io/docs/concepts/configuration/secret/
+		// These secrets contain the "data"-key before the actual identifier "kind: Secret"
+		"kubernetes.yaml": "apiVersion: v1 data: extra: YmFyCg== kind: Secret metadata: name: secret-sa-sample annotations: kubernetes.io/service-account.name: 'sa-name'",                                                                                                                // gitleaks:allow
```

Comment:
It would be good to add test cases that have quoted yaml (`kind: "Secret"`) and use 'actual' newlines between entries instead of spaces.

This can be accomplish by using `\n` inside of double-quotes, or using backticks (`) and formatting text as-is.

Reply:
You are correct about the quoting. Your suggested regex in the above comment will fail if the secret is qouted like kind: 'Secret' 

See here for an example https://regex101.com/r/VThyQ1/1

I've adjusted the regex to match quotes and added the newlines